src/random.cr

```diff
@@ -1,66 +1,126 @@
 require "random/mt19937"
 
+# `Random` provides an interface for random values generation, using a pseudo random number generator (PRNG).
+# 
+# ```
+# Random.new_seed # => 112705036
+# Random.rand     # => 0.167595
+# Random.rand(5)  # => 2
+# ```
+#
+# The above methods delegate to a `Random` instance.
+#
+# ```
+# r = Random.new
+# r.rand      # => 0.0372991
+# r.next_bool # => true
+# r.next_int  # => 2223112
+# ```
+#
+# This module also defines a global method `#rand`, which `Array#sample` and `Array#shuffle` delegates.
+#
+# ```
+# rand     # => 0.293829
+# rand(10) # => 8
+# ```
 module Random
   DEFAULT = MT19937.new
 
+  # Generates a `UInt32` based on current time using `read_cycle_counter` from LLVM.
```

Comment:
I think how the seed is obtained is quite relevant to a PRNG. That it's an LLVM intrinsic is indeed irrelevant though, instead some short pointer as to what the "cycle counter" is would be good.